frontend/src/hooks/useLastViewedProject.ts

```diff
@@ -9,9 +10,14 @@ export const useLastViewedProject = () => {
         return getLocalStorageItem(key);
     });
 
+    const { emitEvent } = useCustomEvent('lastViewedUpdated', () => {
+        setLastViewed(getLocalStorageItem(key));
```

Comment:
when some other hook updated we update our local last viewed state from the local storage